src/bicep/modules/remote-access.bicep

```diff
@@ -48,99 +51,104 @@ param windowsVmSku string
+      tags: tags
+    }
   }
-}
 
-module linuxNetworkInterface '../modules/network-interface.bicep' = {
-  name: 'remoteAccess-linuxNetworkInterface'
-  scope: resourceGroup(hubProperties.subscriptionId, hubProperties.resourceGroupName)
-  params: {
-    ipConfigurationName: hubProperties.linuxNetworkInterfaceIpConfigurationName
-    location: location
-    mlzTags: mlzTags
-    name: hubProperties.linuxNetworkInterfaceName
-    networkSecurityGroupId: hubNetworkSecurityGroupResourceId
-    privateIPAddressAllocationMethod: linuxNetworkInterfacePrivateIPAddressAllocationMethod
-    subnetId: hubSubnetResourceId
-    tags: tags
+module linuxNetworkInterface '../modules/network-interface.bicep' =
+  if (deployLinuxVirtualMachine) {
+    name: 'remoteAccess-linuxNetworkInterface'
+    scope: resourceGroup(hubProperties.subscriptionId, hubProperties.resourceGroupName)
+    params: {
+      ipConfigurationName: hubProperties.linuxNetworkInterfaceIpConfigurationName
+      location: location
+      mlzTags: mlzTags
+      name: hubProperties.linuxNetworkInterfaceName
+      networkSecurityGroupId: hubNetworkSecurityGroupResourceId
+      privateIPAddressAllocationMethod: linuxNetworkInterfacePrivateIPAddressAllocationMethod
+      subnetId: hubSubnetResourceId
+      tags: tags
+    }
   }
-}
 
-module linuxVirtualMachine '../modules/linux-virtual-machine.bicep' = {
-  name: 'remoteAccess-linuxVirtualMachine'
-  scope: resourceGroup(hubProperties.subscriptionId, hubProperties.resourceGroupName)
-  params: {
-    adminPasswordOrKey: linuxVmAdminPasswordOrKey
-    adminUsername: linuxVmAdminUsername
-    authenticationType: linuxVmAuthenticationType
-    diskEncryptionSetResourceId: diskEncryptionSetResourceId
-    diskName: hubProperties.linuxDiskName
-    location: location
-    logAnalyticsWorkspaceId: logAnalyticsWorkspaceId
-    mlzTags: mlzTags
-    name: hubProperties.linuxVmName
-    networkInterfaceName: linuxNetworkInterface.outputs.name
-    osDiskCreateOption: linuxVmOsDiskCreateOption
-    osDiskType: linuxVmOsDiskType
-    tags: tags
-    vmImageOffer: linuxVmImageOffer
-    vmImagePublisher: linuxVmImagePublisher
-    vmImageSku: linuxVmImageSku
-    vmImageVersion: linuxVmImageVersion
-    vmSize: linuxVmSize
+module linuxVirtualMachine '../modules/linux-virtual-machine.bicep' =
+  if (deployLinuxVirtualMachine) {
+    name: 'remoteAccess-linuxVirtualMachine'
+    scope: resourceGroup(hubProperties.subscriptionId, hubProperties.resourceGroupName)
+    params: {
+      adminPasswordOrKey: linuxVmAdminPasswordOrKey
+      adminUsername: linuxVmAdminUsername
+      authenticationType: linuxVmAuthenticationType
+      diskEncryptionSetResourceId: diskEncryptionSetResourceId
+      diskName: hubProperties.linuxDiskName
+      location: location
+      logAnalyticsWorkspaceId: logAnalyticsWorkspaceId
+      mlzTags: mlzTags
+      name: hubProperties.linuxVmName
+      networkInterfaceName: linuxNetworkInterface.outputs.name
```

Comment:
@tsweatman1 line 102 needs a condition to fix your deployment error, like so:
networkInterfaceName: deployLinuxVirtualMachine ? linuxNetworkInterface.outputs.name : ''